x-pack/plugins/security_solution/server/lib/detection_engine/schemas/rule_converters.ts

```diff
@@ -386,64 +390,69 @@ const shouldUpdateVersion = (params: PatchRulesSchema): boolean => {
-      ruleNameOverride: params.rule_name_override ?? existingParams.ruleNameOverride,
-      setup: params.setup ?? existingParams.setup,
-      severity: params.severity ?? existingParams.severity,
-      severityMapping: params.severity_mapping ?? existingParams.severityMapping,
-      threat: params.threat ?? existingParams.threat,
-      timestampOverride: params.timestamp_override ?? existingParams.timestampOverride,
+      license: nextParams.license ?? existingParams.license,
+      outputIndex: nextParams.output_index ?? existingParams.outputIndex,
+      timelineId: nextParams.timeline_id ?? existingParams.timelineId,
+      timelineTitle: nextParams.timeline_title ?? existingParams.timelineTitle,
+      meta: nextParams.meta ?? existingParams.meta,
+      maxSignals: nextParams.max_signals ?? existingParams.maxSignals,
+      relatedIntegrations: nextParams.related_integrations ?? existingParams.relatedIntegrations,
+      requiredFields: nextParams.required_fields ?? existingParams.requiredFields,
+      riskScore: nextParams.risk_score ?? existingParams.riskScore,
+      riskScoreMapping: nextParams.risk_score_mapping ?? existingParams.riskScoreMapping,
+      ruleNameOverride: nextParams.rule_name_override ?? existingParams.ruleNameOverride,
+      setup: nextParams.setup ?? existingParams.setup,
+      severity: nextParams.severity ?? existingParams.severity,
+      severityMapping: nextParams.severity_mapping ?? existingParams.severityMapping,
+      threat: nextParams.threat ?? existingParams.threat,
+      timestampOverride: nextParams.timestamp_override ?? existingParams.timestampOverride,
       timestampOverrideFallbackDisabled:
-        params.timestamp_override_fallback_disabled ??
+        nextParams.timestamp_override_fallback_disabled ??
         existingParams.timestampOverrideFallbackDisabled,
-      to: params.to ?? existingParams.to,
-      references: params.references ?? existingParams.references,
-      namespace: params.namespace ?? existingParams.namespace,
-      note: params.note ?? existingParams.note,
+      to: nextParams.to ?? existingParams.to,
+      references: nextParams.references ?? existingParams.references,
+      namespace: nextParams.namespace ?? existingParams.namespace,
+      note: nextParams.note ?? existingParams.note,
       // Always use the version from the request if specified. If it isn't specified, leave immutable rules alone and
       // increment the version of mutable rules by 1.
-      version:
-        params.version ?? existingParams.immutable
-          ? existingParams.version
-          : shouldUpdateVersion(params)
-          ? existingParams.version + 1
-          : existingParams.version,
-      exceptionsList: params.exceptions_list ?? existingParams.exceptionsList,
+      version: nextParams.version
+        ? nextParams.version
+        : shouldUpdateVersion(nextParams, existingParams)
+        ? existingParams.version + 1
+        : existingParams.version,
```

Comment:
It looks like currently, our users are able to modify the rule version of prebuilt rules via CRUD endpoints. I think we need to create some follow-up tickets for disallowing that. Version updates should be done internally and controlled by the app on the server side.

Also, I believe this logic of incrementing the version should move to the routes like the `add_prepackaged_rules_route.ts` one. Here in the `convertPatchAPIToInternalSchema` it would become `version: nextParams.version ?? existingParams.version`. We can leave it as is in this PR.

Reply:
Ideally, we should disallow users to modify prebuilt rules via API. But given that we're working actively on the prebuilt rule customization epic and the immutability flag will be removed, probably it's not worth spending time to add extra validation to the routes 🤷‍♂️